src/metrics/runner.rs

```diff
@@ -54,12 +62,29 @@ impl MetricsRunner {
         )
         .unwrap();
 
+        let total_api_boundary_nodes = register_int_gauge_with_registry!(
```

Comment:
Nit: maybe we can squeeze it into a single metric with a label `healthy` `yes/no` or smth like that? Not critical, but maybe would make things simpler.

Reply:
Btw should i add metrics to the cache? I guess You envisioned to cache all the metrics, no?

Reply:
Also another point. In the next PR i wanted to add a flag `k-top-nodes` to the `ic-gateway` (i.e. dynamic routing to `k` nodes with best score). In this case we will display only `k` nodes as healthy. Won't it be confusing?

Reply:
we could also extend the [RoutesStats](https://github.com/dfinity/agent-rs/blob/96910e70c44946d4ba8a78f050fc33610e1dda60/ic-agent/src/agent/route_provider.rs#L41C12-L41C23):
```
pub struct RoutesStats {
    pub total_nodes: usize,
    pub healthy_nodes: Option<usize>,
    pub nodes_in_routing: usize,
}
```

Reply:
Yeah something like this with a label. It's how it's done in other places e.g. healthy nodes in `ic-boundary` etc I think.

You don't need to cache anything, it just generates the response HTTP body from the metrics and stores it in the ArcSwap once every few seconds. This helps to avoid doing that on every scrape.

About `RoutesStats` I don't get it. On the route provider we have `n_ordered_routes()` method which returns top-n nodes, right? What does it has to do with router stats, it just should show total/healthy nodes, irrespective of what we ask it,  no?

Reply:
done

Reply:
this is not relevant for this PR, we can discuss it later

Reply:
Ok